### ISSUE TITLE
skip sanity check

### ISSUE THREAD

**Issue by kometchtech** [easybuilders/easybuild-framework#2288]
Hi, developer

When trying to build GCCcore-4.9.3 in the following environment, it seems necessary to check the library related to libsanitizer in the sanity check step, but in case of aarch64 environment libsanitizer support becomes no , The library is not generated.

`checking for libsanitizer support... no`

**gcc.py**
```
521         if LooseVersion(self.version) >= LooseVersion('4.2'):
522             # libgomp was added in GCC 4.2.0
523             ["libgomp.%s" % sharedlib_ext, "libgomp.a"]
524         if os_type == 'Linux':
525             lib_files.extend(["libgcc_s.%s" % sharedlib_ext])
526             # libmudflap is replaced by asan (see release notes gcc 4.9.0)
527             if LooseVersion(self.version) < LooseVersion("4.9.0"):
528                 lib_files.extend(["libmudflap.%s" % sharedlib_ext, "libmudflap.a"])
529             else:
530                 lib_files.extend(["libasan.%s" % sharedlib_ext, "libasan.a"])
```
It seems that #1626  has been argued in the past, but do you make a branch judgment in gcc.py or will you be able to skip the sanity check in the first place?

```
$ cat /etc/redhat-release
CentOS Linux release 7.2.1603 (AltArch)
$ uname -m -p -i
aarch64 aarch64 aarch64
```
Best regards.

**Comment by boegel** [comment]
@kometchtech There is no easy way to skip the sanity check entirely right now. We're a bit reluctant to allow this, because the sanity check is there for good reason, it protects you from ending up with a partial installation.

Long-term, we should indeed update the GCC easyblock (`gcc.py`) to not check for `libsanitizer` on AARCH64 (@geimer: why didn't you run into this?).

As a temporary workaround, you can customise the location that are checked during the sanity 
check by including something like this in the easyconfig file for GCC you are using (see also http://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#sanity-check):

```python
sanity_check_paths = {
    'files': ['bin/gcc'],
    'dirs': [],
}
```

I strongly recommend to check more than just `bin/gcc` though, this is just an example.
You can determine the full list of locations that are checked by default using `eb -x GCC...eb`, and copy those locations in your easyconfig file (except for `libsanitizer`).

To finalise the installation after adding a custom `sanity_check_paths`, you can just run `eb --module-only GCC..eb`, to avoid reinstalling GCC from scratch again, see also http://easybuild.readthedocs.io/en/latest/Partial_installations.html#module-only.

Let us know if you have any more questions on this.

**Comment by kometchtech** [comment]
@boegel 

Thank you for your polite answer.
For the sanity_check_paths parameter, I thought that it would be added to existing ones. In fact it is overwritten.
As you say, it is a confirmation that the file is installed correctly, so it is a bad way to handle comments that I did.
I will try it tomorrow.

Best regards.

**Comment by geimer** [comment]
@boegel: I haven't built anything older than GCC 6 on my AARCH64 box (IMHO building ancient compilers is a waste of time and energy), which does provide the sanitizer libraries.  So maybe we need a platform-specific version check in the EasyBlock?

**Comment by kometchtech** [comment]
@boegel 

Contact became late.

I was able to confirm that sanity check itself can be overwritten by appending the parameter of sanity_check_paths to the eb file for the time being.
```
sanity_check_paths = {
    'files': [
        'bin/gcc', 'bin/c++', 'bin/cpp', 'bin/g++', 'bin/gcov', 'bin/gfortran', 'lib64/libgcc_s.so', 'lib64/libgfortran.a',
        'lib64/libgfortran.so', 'lib64/libstdc++.a', 'lib64/libstdc++.so', 'libexec/gcc/aarch64-unknown-linux-gnu/4.9.3/liblto_plugin.so',
        'libexec/gcc/aarch64-unknown-linux-gnu/4.9.3/lto-wrapper', 'libexec/gcc/aarch64-unknown-linux-gnu/4.9.3/lto1'
    ],
    'dirs': [
        'include/c++/4.9.3', 'lib/gcc/aarch64-unknown-linux-gnu/4.9.3'
    ],
}
```


Permanently, consider adding a conditional branch to gcc.py.

Thank you very much.

**Comment by boegel** [comment]
@geimer Any idea since which version the sanitizer libraries are available on AARCH64?

**Comment by boegel** [comment]
Although *not* recommended, support for skipping the sanity check was implemented in #2549 (included since EasyBuild v3.7.0), so closing this issue...
